pkg/sst/sst.go

```diff
@@ -234,3 +271,163 @@ func getBits(val, i, j uint32) uint32 {
 func isBitSet(val, n uint32) bool {
 	return val&(1<<n) != 0
 }
+
+func setBit(val, n uint32) uint32 {
+	return val | (1 << n)
+}
+
+func clearBit(val, n uint32) uint32 {
+	return val &^ (1 << n)
+}
+
+func setBFStatus(info *SstPackageInfo, status bool) error {
+	rsp, err := sendMboxCmd(info.pkg.cpus[0], CONFIG_TDP, CONFIG_TDP_GET_TDP_CONTROL, uint32(info.PPCurrentLevel))
+	if err != nil {
+		return fmt.Errorf("failed to read SST status: %v", err)
+	}
+
+	req := clearBit(rsp, 17)
+	if status {
+		req = setBit(rsp, 17)
+	}
+
+	if _, err = sendMboxCmd(info.pkg.cpus[0], CONFIG_TDP, CONFIG_TDP_SET_TDP_CONTROL, req); err != nil {
+		return fmt.Errorf("failed to enable SST %s: %v", "BF", err)
+	}
+
+	info.BFEnabled = status
+
+	return nil
+}
+
+func adjustScalingMinFromBaseFreq(cpu utils.ID) error {
+	minScalingFreq, err := getCPUScalingMinFreq(cpu)
+	if err != nil {
+		sstlog.Debugf("sst-bf cpu %d set scaling freq error: %v", cpu, err)
+		return err
+	}
+
+	baseFreq, err := getCPUBaseFreq(cpu)
+	if err != nil {
+		sstlog.Debugf("sst-bf cpu %d set base freq error: %v", cpu, err)
+		return err
+	}
+
+	// Boost the scaling_min_freq if the base frequency is higher
+	if minScalingFreq < baseFreq {
+		if err := setCPUScalingMinFreq(cpu, baseFreq); err != nil {
+			sstlog.Debugf("sst-bf cpu %d set min scaling freq error: %v", cpu, err)
+			return err
+		}
+	}
+
+	return nil
+}
+
+func setScalingMin2CPUInfoMax(info *SstPackageInfo) error {
+	for _, cpu := range info.pkg.cpus {
+		err := setCPUScalingMin2CPUInfoMaxFreq(cpu)
+		if err != nil {
+			sstlog.Debugf("sst-bf cpu %d set min freq error: %v", cpu, err)
+			return err
```

Comment:
Hmm, does the separate debug print have some value here. I'd either include description in the error message

```suggestion
			return fmt.Errorf("sst-bf cpu %d set min freq error: %w", cpu, err)
```

OR alternatively just passthrough the error

```suggestion
			return err
```


Ditto other similar cases in the PR